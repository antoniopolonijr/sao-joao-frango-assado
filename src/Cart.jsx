const intl = new Intl.NumberFormat("pt-BR", {
  style: "currency",
  currency: "BRL",
});

export default function Cart({ cart, checkout }) {
  let total = 0;
  for (let i = 0; i < cart.length; i++) {
    const current = cart[i];
    total += current.food.sizes[current.size];
  }
  return (
    <div className="cart">
      <h2>Carrinho</h2>
      <ul>
        {cart.map((item, index) => (
          <li key={index}>
            <span className="type">{item.food.name}</span> –{" "}
            <span className="size">{item.size}</span> –{" "}
            <span className="price">{item.price}</span>
          </li>
        ))}
      </ul>
      <p className="cart-total">Total: {intl.format(total)}</p>
      <button onClick={checkout}>Finalizar a Compra</button>
    </div>
  );
}
