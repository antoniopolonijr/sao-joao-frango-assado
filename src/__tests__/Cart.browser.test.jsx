import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import Cart from "../Cart";

test("snapshot with nothing in cart", () => {
  const { asFragment } = render(<Cart cart={[]} />);
  expect(asFragment()).toMatchSnapshot();
});

test("snapshot with some stuff in cart", () => {
  const { asFragment } = render(
    <Cart
      cart={[
        {
          food: {
            id: "frango_assado",
            name: "Frango",
            category: "Carne",
            description: "Frango Assado com Farofa",
            image: "/public/foods/frango_assado.webp",
            sizes: {
              P: 0,
              M: 33,
              G: 50,
            },
          },
          size: "G",
          price: "R$ 50,00",
        },
        {
          food: {
            id: "nhoque",
            name: "Nhoque",
            category: "Massa",
            description: "Nhoque de Batata Ã  Bolonhesa",
            image: "/public/foods/nhoque.webp",
            sizes: {
              P: 0,
              M: 24,
              G: 48,
            },
          },
          size: "M",
          price: "R$ 24,00",
        },
        {
          food: {
            id: "maionese",
            name: "Maionese",
            category: "Acompanhamento",
            description: "Salada de Maionese com Legumes",
            image: "/public/foods/maionese.webp",
            sizes: {
              P: 0,
              M: 24,
              G: 0,
            },
          },
          size: "M",
          price: "R$ 24,00",
        },
      ]}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
