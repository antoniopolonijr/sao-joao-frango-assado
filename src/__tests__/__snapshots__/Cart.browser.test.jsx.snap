// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot with nothing in cart 1`] = `
<DocumentFragment>
  <div
    class="cart"
  >
    <h2>
      Carrinho
    </h2>
    <ul />
    <p>
      Total: R$ 0,00
    </p>
    <button>
      Finalizar a Compra
    </button>
  </div>
</DocumentFragment>
`;

exports[`snapshot with some stuff in cart 1`] = `
<DocumentFragment>
  <div
    class="cart"
  >
    <h2>
      Carrinho
    </h2>
    <ul>
      <li>
        <span
          class="type"
        >
          Frango
        </span>
         – 
        <span
          class="size"
        >
          G
        </span>
         – 
        <span
          class="price"
        >
          R$ 50,00
        </span>
      </li>
      <li>
        <span
          class="type"
        >
          Nhoque
        </span>
         – 
        <span
          class="size"
        >
          M
        </span>
         – 
        <span
          class="price"
        >
          R$ 24,00
        </span>
      </li>
      <li>
        <span
          class="type"
        >
          Maionese
        </span>
         – 
        <span
          class="size"
        >
          M
        </span>
         – 
        <span
          class="price"
        >
          R$ 24,00
        </span>
      </li>
    </ul>
    <p>
      Total: R$ 98,00
    </p>
    <button>
      Finalizar a Compra
    </button>
  </div>
</DocumentFragment>
`;
